vclib/src/commonMain/kotlin/at/asitplus/wallet/lib/data/dif/PresentationSubmissionValidator.kt

```diff
@@ -0,0 +1,89 @@
+package at.asitplus.wallet.lib.data.dif
+
+import at.asitplus.KmmResult
+import kotlinx.serialization.Serializable
+
+@Serializable
+sealed interface PresentationSubmissionValidator {
+    companion object {
+        @Throws(MissingInputDescriptorGroupException::class)
+        fun createInstance(
+            submissionRequirements: Collection<SubmissionRequirement>?,
+            inputDescriptors: Collection<InputDescriptor>,
+        ): KmmResult<PresentationSubmissionValidator> {
+            val verifier = submissionRequirements?.let { _ ->
+                SubmissionRequirementsValidator(
+                    submissionRequirements = submissionRequirements,
+                    inputDescriptorGroups = inputDescriptors.associate {
+                        it.id to (it.group ?: return KmmResult.failure(
+                            MissingInputDescriptorGroupException(it)
+                        ))
+                    },
+                )
+            } ?: InputDescriptorSubmissionsValidator(
+                inputDescriptorIds = inputDescriptors.map { it.id }.toSet()
+            )
+            return KmmResult.success(verifier)
+        }
+    }
+
+    /**
+     * Checks, whether all submission requirements are satisfied
+     */
+    fun isSubmissionRequirementsSatisfied(
+        submittedInputDescriptorIds: Set<String>,
+    ): Boolean
+
+    /**
+     * Checks, whether submission requirements are satisfied, and also fails if there are unnecessary submissions
+     */
+    fun isValidSubmission(
+        submittedInputDescriptorIds: Set<String>,
+    ): Boolean {
+        return isSubmissionRequirementsSatisfied(submittedInputDescriptorIds) && findUnnecessaryInputDescriptorSubmissions(
```

Comment:
Please refactor to expression body

Reply:
fixed in commit 65b06496293f5ec3ea9babe119596f8a137085d8